### PR TITLE
[add-skip-ci-mq-check] Add skip ci check

### PR DESCRIPTION
**Pull request by CelianR**
### What does this PR do?

Adds a skip ci check to the merge queue

### CODE REVIEW

File: .gitlab/common/skip_ci_check.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Disallow [skip ci] in commit messages
+
+skip-ci-check:
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  needs: []
+  stage: setup
+  tags: ["arch:amd64"]
+  rules:
+    - !reference [.except_mergequeue]
```
Comment (L3n41c) [inline .gitlab/common/skip_ci_check.yml:9]: Isn’t this the opposite of what is in the PR’s description:
> ### What does this PR do?
> Adds a skip ci check to the merge queue

With these rules, this check will run everywhere except on the merge queue.

File: .gitlab/common/skip_ci_check.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  script:
+    - commit_message="$(git log --format=%B -n 1 $CI_COMMIT_SHA)"
+    # https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs
+    - |
+      case "$commit_message" in
+          *"[ci skip]"*|*"[skip ci]"*|*"[actions skip]"*|*"[skip actions]"*|*"[no ci]"*|*"skip-checks: true"*|*"skip-checks:true"*)
+              echo "error: The commit message of this PR contains ci skip tags. Do not skip checks when merging PRs, please change the description of the merge commit" >& 2
+              exit 1
+          ;;
+      esac
```
Comment (L3n41c) [inline .gitlab/common/skip_ci_check.yml:20]:
```suggestion
      if [[ "$commit_message" =~ *"[ci skip]"*|*"[skip ci]"*|*"[actions skip]"*|*"[skip actions]"*|*"[no ci]"*|*"skip-checks: true"*|*"skip-checks:true"* ]]; then
            echo "error: The commit message of this PR contains ci skip tags. Do not skip checks when merging PRs, please change the description of the merge commit" >& 2
            exit 1
      fi
```

**Review by L3n41c — COMMENTED** [review]
Am I correct to think that this is a “only a nice-to-have” because, anyways, today, a PR with `[ci skip]` in his comment would already not be mergeable because there are some GitHub Actions jobs that are marked as `[Required]` ?

**Review by chouetz — APPROVED** [review]

**Pull request merged**